pkg/daemon/ceph/client/pool.go

```diff
@@ -447,6 +465,9 @@ func createReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 		// Create the pool since it doesn't exist yet
 		// If there was some error other than ENOENT (not exists), go ahead and ensure the pool is created anyway
 		args := []string{"osd", "pool", "create", pool.Name, pgCount, "replicated", crushRuleName, "--size", strconv.FormatUint(uint64(pool.Replicated.Size), 10)}
+		if autoscaleMode != "" {
+			args = append(args, "--autoscale-mode", autoscaleMode)
+		}
```

Comment:
Importantly, if autoscale mode is not set and has previously been set, Rook should change the autoscale mode to the default one used by Ceph. It is impossible for Rook to know when the mode has been changed, so if the mode is `""` (default), Rook should set the mode to the default mode. 

I can't find in the Ceph docs how to cause this to happen explicitly. Because Ceph is usually not an idempotent desired-state service, Rook will likely need to look up the current default autoscale mode for the cluster and set that on the pool explicitly.

Perhaps @ideepika would know if setting the autoscale mode to `""` would cause Ceph to pull the default value.